File: Quick Sort/QuickSort.js
function swap(array, i, j) {
    let temp = array[i];
    array[i] = array[j];
    array[j] = temp;
}

function partition(array, head, tail) {
    let pivot = array[tail];
    let i = head - 1;
    for (let j = head; j < tail; j++) {
        if (array[j] < pivot) {
            i++;
            swap(array, i, j);
        }
    }
    swap(array, i + 1, tail);
    return i + 1;
}

function quickSort(array, head, tail) {
    if (head < tail) {
        let pivotIndex = partition(array, head, tail);
        quickSort(array, head, pivotIndex - 1);
        quickSort(array, pivotIndex + 1, tail);
    }
}

function showResult() {
    let arr = [3, 1, 8, 6, 9, 2, 7];
    quickSort(arr, 0, arr.length - 1);
    let result = "";
    for (let i = 0; i < arr.length; i++) {
        result += arr[i] + " ";
    }
    document.getElementById("demo").innerHTML = result;
}

File: Quick Sort/QuickSort.cpp
#include<iostream>

using namespace std;
int* QuickSort(int *Arr, int head, int tail){
    if(tail<=head){
        return Arr;
    }
    int pivot=Arr[tail];
    int i=head-1, j=head;
    while(Arr[j]!=pivot){
        if(Arr[j]<pivot){
            i++;
            int temp=Arr[j];
            Arr[j]=Arr[i];
            Arr[i]=temp;
            j++;
        }else{
            j++;
        }
    }
    int temp=Arr[i];
    Arr[i]=pivot;
    Arr[tail]=temp;
    int *LeftArray=QuickSort(Arr, head, i-1);
    int *RightArray=QuickSort(Arr, i+1, tail);
}
int main(){
    int Array[10]={5,6,9,8,1,3,4,0,2,7};
    cout<<QuickSort(Array, 0, 9);
}